src/ast/irbuilderbpf.cpp

```diff
@@ -812,5 +913,93 @@ Value *IRBuilderBPF::CreatKFuncArg(Value *ctx,
+  assert(return_value && return_value->getType() == getInt32Ty());
+  if (bpftrace_.disable_helper_check_)
+    return;
+  if (!bpftrace_.check_all_helper_ &&
+      (is_probe_read_funcs(func_id) ||
+       func_id == libbpf::BPF_FUNC_map_lookup_elem))
+    return;
+
+  int error_id = helper_error_id_++;
+  bpftrace_.helper_error_info_[error_id] = { .func_id = func_id, .loc = loc };
+
+  auto elements = AsyncEvent::HelperError().asLLVMType(*this);
+  StructType *helper_error_struct = GetStructType("helper_error_t",
+                                                  elements,
+                                                  true);
+  AllocaInst *buf = CreateAllocaBPF(helper_error_struct, "helper_error_t");
+  CreateStore(GetIntSameSize(asyncactionint(AsyncAction::helper_error),
+                             elements.at(0)),
+              CreateGEP(buf, { getInt64(0), getInt32(0) }));
+  CreateStore(GetIntSameSize(error_id, elements.at(1)),
+              CreateGEP(buf, { getInt64(0), getInt32(1) }));
+  CreateStore(return_value, CreateGEP(buf, { getInt64(0), getInt32(2) }));
+
+  auto &layout = module_.getDataLayout();
+  auto struct_size = layout.getTypeAllocSize(helper_error_struct);
+  CreatePerfEventOutput(ctx, buf, struct_size);
+  CreateLifetimeEnd(buf);
+}
+
+// Report error if a return value < 0 (or return value == 0 if compare_zero is
+// true)
+void IRBuilderBPF::CreateHelperErrorCond(Value *ctx,
+                                         Value *return_value,
+                                         libbpf::bpf_func_id func_id,
+                                         const location &loc,
+                                         bool compare_zero)
+{
+  assert(ctx && ctx->getType() == getInt8PtrTy());
+  if (bpftrace_.disable_helper_check_)
+    return;
+  if (!bpftrace_.check_all_helper_ &&
+      (is_probe_read_funcs(func_id) ||
+       func_id == libbpf::BPF_FUNC_map_lookup_elem))
+    return;
+
+  Function *parent = GetInsertBlock()->getParent();
+  BasicBlock *helper_success_block = BasicBlock::Create(module_.getContext(),
+                                                        "helper_success",
+                                                        parent);
+  BasicBlock *helper_failure_block = BasicBlock::Create(module_.getContext(),
+                                                        "helper_failure",
+                                                        parent);
+  BasicBlock *helper_merge_block = BasicBlock::Create(module_.getContext(),
+                                                      "helper_merge",
+                                                      parent);
+  auto *ret = CreateIntCast(return_value, getInt32Ty(), true);
+  Value *condition;
+  if (compare_zero)
+    condition = CreateICmpNE(ret, Constant::getNullValue(ret->getType()));
+  else
+    condition = CreateICmpSGE(ret, Constant::getNullValue(ret->getType()));
+  CreateCondBr(condition, helper_success_block, helper_failure_block);
```

Comment:
```suggestion
  CreateCondBr(condition, helper_merge_block, helper_failure_block);
```
Don't think the success block is needed.